src/loader.rs

```diff
@@ -0,0 +1,87 @@
+extern crate libloading;
+use self::libloading::{Library, Symbol};
+use super::driver::{Driver, DriverSchema};
+use std::collections::HashMap;
+
+pub type Result<T> = ::std::io::Result<T>;
+
+/// Trait implemented by anthing that wishes to load a driver at runtime.
+pub trait DriverLoader {
+  fn get_driver<'a>(&self) -> Option<&'a Driver>;
+  fn get_schema<'a>(&self) -> Option<&'a DriverSchema>;
```

Comment:
I know, I don't like it but from what I see we would have to add every possible trait there.
Also this is being re-written if you have not seen that none of the boxes are checked